cmd/clusterctl/pkg/client/cluster/objectgraph.go

```diff
@@ -45,35 +43,45 @@ type dependAttributes struct {
 type node struct {
 	identity corev1.ObjectReference
 
-	// dependents contains the list of nodes that are owned by the current node.
-	// Nb. This is the reverse of metadata.OwnerReferences.
-	dependents map[*node]dependAttributes
+	// owners contains the list of nodes that are owned by the current node.
+	owners map[*node]ownerReferenceAttributes
```

Comment:
I wonder if we can merge the owners and softOwners maps by adding a field to `ownerReferenceAttributes` called `implicit` or something similar?

Reply:
I would prefer to keep the two concepts separated because one mirrors what is in the Kubernetes API while the other is a Cluster API convention, but If you prefer I can merge them together.

Reply:
It's ok, we can clean it up later